include/boost/beast/http/impl/read.ipp

```diff
@@ -283,36 +296,23 @@ operator()(
     error_code ec,
     std::size_t bytes_transferred)
 {
-    switch(state_)
+    BOOST_ASIO_CORO_REENTER(*this)
     {
-    case 0:
-        if(Condition{}(p_))
+        if (Condition{}(p_))
         {
-            state_ = 1;
-            return s_.get_io_service().post(
+            BOOST_ASIO_CORO_YIELD s_.get_io_service().post(
```

Comment:
The problem with putting this all on one line is that it is harder to set breakpoints with visual studio, and it also moves the important part farther to the right in the editor. I try to keep the code narrow, it helps when I screenshare and do collaborative code reviews.